HIP/hip-0000-introduction-of-blocks.md

```diff
@@ -0,0 +1,199 @@
+  | nonce | Mirror Node | Hardcoded by Mirror Node(s) to hex-encoded 0 |
+  | parentHash | Mirror Node | Computed by Mirror Node(s). The hash of the parent block. |
+  | receiptsRoot | Mirror Node | (???) |
+  | sha3Uncles | Mirror Node | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array (0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347) |
+  | size | Mirror Node | Computed by Mirror Node(s). The sum of the bytes of all transaction(s) of type ContractLocalCall, ContractCall and ContractCreate within the block. transaction is the transaction specified in the RecordStreamObject and not the Transaction Record |
+  | stateRoot | Mirror Node | (???) |
+  | totalDifficulty | Mirror Node | Hardcoded by Mirror Nodes(s) to hex-encoded 0 |
+  | transactions | Mirror Node | Computed by Mirror Node(s) by ingesting the record stream and aggregating RecordStreamObjects of type ContractLocalCall, ContractCall and ContractCreate within the block. |
+  | transactionsRoot | Mirror Node | (???) |
+  | uncles | Mirror Node | Hardcoded by Mirror Node(s) to empty array [] |
+
+## Backwards Compatibility
+
+The following breaking changes will be introduced with the implementation of the HIP:
+
+- `BLOCKHASH` will no longer return an empty `hash` but the actual `hash` of the block as per the EVM specification.
+- `NUMBER` will no longer return the `timestamp` of the `block`, but rather the proper `block` number as specified in the HIP.
+
+## Security Implications
+
+- The specification does not have any security implications.
+
+## How to Teach this
+
+- Respective documentation will be added.
+
+## Reference Implementation
+
+Initial POC in `services`:
+
+- [https://github.com/hashgraph/hedera-services/commit/04501de76f3e649b3f741dd2453c05e2631e7393#diff-1ca5738f0fe42a25f91d641aea2104dd51bec20398ce6c2aa18afde3d5a590d9R417](https://github.com/hashgraph/hedera-services/commit/04501de76f3e649b3f741dd2453c05e2631e7393#diff-1ca5738f0fe42a25f91d641aea2104dd51bec20398ce6c2aa18afde3d5a590d9R417)
+
+## Rejected Ideas
+
+Two iterations were conducted prior to settling on this approach.
+
+1. Rounds as blocks - the first iteration was proposing that each block is the `consensus round`. It became clear that this approach is not suitable as it implied that multiple blocks are to be issued per second. That would add additional load to infrastructure providers when clients are iterating and going through blocks to query for data.
+- Events - the second iteration suggested that `consensus events` are to be defined as `blocks`. The proposal had a much higher cognitive load (1 block = 1 record file is easier to grasp) and it required more changes to the `platform` in order to be implemented. To add on top of that, the same drawback as the `high frequency` blocks was present as well.
+
+## Open Issues
+
+- What will be the considered `baseFeePerGas` property for blocks? There are different pricing for `Create`, `Call` and `LocalCall` transactions.
```

Comment:
I think we lock it in at zero until we need to use it.  This is a congestion related aspect of Ethereum's block production process (Specificallt EIP-1559), which we don't do.